playground/src/lib/hooks/useOpenApiSpec.ts

```diff
@@ -12,19 +13,15 @@ export function openApiSpecQueryOptions(openapi: OpenApiContext | undefined) {
     queryKey: OPENAPI_CACHE_KEY,
     queryFn: async () => {
       if (openapi?.content) {
-        return JSON.stringify(openapi.content);
+        return openapi.content;
       }
 
       if (!openapi?.url) {
         return undefined;
       }
 
-      const response = await fetch(openapi.url);
-      if (!response.ok) {
-        throw new Error(`HTTP error! status: ${response.status}`);
-      }
-
-      return response.text();
+      const stringifiedSpec = await api.getOpenApiSpec(openapi.url);
```

Comment:
use an api helper that will throw custom errors, that in turn the frontend knows how to render with helpful error messages